modules/dnn/src/layers/detection_output_layer.cpp

```diff
@@ -611,8 +611,14 @@ class DetectionOutputLayerImpl CV_FINAL : public DetectionOutputLayer
                 }
             }
             // Keep outputs k results per image.
-            std::sort(scoreIndexPairs.begin(), scoreIndexPairs.end(),
-                      util::SortScorePairDescend<std::pair<int, int> >);
+            // if _keepTopK * 8 <= scoreIndexPairs.size(), apply partial sort.
+            if ((_keepTopK << 3) > scoreIndexPairs.size()) {
```

Comment:
` << 3`

modern compilers are smart enough, so just `* 8` should work well too.